worker/src/handlers/statement/allowedWorkerQueues.js

```diff
@@ -0,0 +1,51 @@
+import logger from 'lib/logger';
+import {
+  STATEMENT_QUERYBUILDERCACHE_QUEUE,
+  STATEMENT_EXTRACT_PERSONAS_QUEUE,
+  STATEMENT_FORWARDING_QUEUE,
+} from 'lib/constants/statements';
+
+/**
+ * allowable Worker Queues
+ *
+ * @type {Map<string, true>}
+ */
+const allowableWorkerQueues = new Map([
+  [STATEMENT_QUERYBUILDERCACHE_QUEUE, true],
+  [STATEMENT_EXTRACT_PERSONAS_QUEUE, true],
+  [STATEMENT_FORWARDING_QUEUE, true],
+]);
+
+/**
+ * @param {string | undefined} allowedWorkerQueuesString
+ * @return {Map<string, true>}
+ * @throws {Error}
+ */
+const getAllowedWorkerQueues = (allowedWorkerQueuesString) => {
+  if (allowedWorkerQueuesString === undefined) {
+    return allowableWorkerQueues;
+  }
+
+  if (allowedWorkerQueuesString === '') {
+    return new Map();
+  }
+
+  return allowedWorkerQueuesString.split(',').reduce(
+    (accMap, queueString) => {
+      if (allowableWorkerQueues.has(queueString)) {
+        return accMap.set(queueString, queueString);
+      }
+      logger.warn(`"${queueString}" is ignored as an allowed worker queue. Allowable worker queues are "${STATEMENT_QUERYBUILDERCACHE_QUEUE}", "${STATEMENT_EXTRACT_PERSONAS_QUEUE}", and "${STATEMENT_FORWARDING_QUEUE}"`);
```

Comment:
Do you think it might be better to map over the `allowableWorkerQueues` to get the queue names and join them by commas to produce the last part of this warning rather than duplicating the list of allowable worker queues for easier maintenance?

Reply:
OK. I just wanted to join the last 2 words with ", and".

Reply:
Ah yeah, I hadn't spotted the "and", happy for us to just display commas though if it's easier (not looked at the changes since this comment just yet).